traits/traits.py

```diff
@@ -1090,15 +1090,16 @@ def define(self, *value_type, **metadata):
                     handler = clone.handler
 
                 if handler is not None:
-                    default_value_type = handler.default_value_type
-                    if default_value_type < 0:
-                        try:
```

Comment:
I'm not happy that this is correct: there _are_ cases in the Traits test suite where this `validate` code is exercised, and where the result is something not equal to the original.